testutil/network/basenet/accounts.go

```diff
@@ -0,0 +1,194 @@
+	if supplierGenesisState == nil {
+		t.Logf(warnNoModuleGenesisFmt, "supplier")
+	} else {
+		// Initialize on-chain accounts for genesis suppliers.
+		memnet.InitSupplierAccountsWithSequence(t, supplierGenesisState.SupplierList...)
+
+	}
+
+	// Retrieve the application module's genesis state from cosmos-sdk in-memory network.
+	appGenesisState := network.GetGenesisState[*apptypes.GenesisState](t, apptypes.ModuleName, memnet)
+	if appGenesisState == nil {
+		t.Logf(warnNoModuleGenesisFmt, "application")
+	} else {
+		// Initialize on-chain accounts for genesis applications.
+		memnet.InitAppAccountsWithSequence(t, appGenesisState.ApplicationList...)
+	}
+
+	// Retrieve the gateway module's genesis state from cosmos-sdk in-memory network.
+	gatewayGenesisState := network.GetGenesisState[*gatewaytypes.GenesisState](t, gatewaytypes.ModuleName, memnet)
+	if gatewayGenesisState == nil {
+		t.Logf(warnNoModuleGenesisFmt, "gateway")
+	} else {
+		// Initialize on-chain accounts for genesis gateways.
+		memnet.InitGatewayAccountsWithSequence(t, gatewayGenesisState.GatewayList...)
+	}
+
+	// need to wait for the account to be initialized in the next block
+	require.NoError(t, memnet.GetNetwork(t).WaitForNextBlock())
+}
+
+// InitAccount initializes an Account by sending it some funds from the validator
+// in the network to the address provided
+func (memnet *BaseInMemoryCosmosNetwork) InitAccount(
+	t *testing.T,
+	addr types.AccAddress,
+) {
+	t.Helper()
+
+	signerAccountNumber := 0
+	// Validator's client context MUST be used for this CLI command because its keyring includes the validator's key
+	clientCtx := memnet.Network.Validators[0].ClientCtx
+	// MUST NOT use memnet.GetClientCtx(t) as its keyring does not include the validator's account
+	// TODO_UPNEXT(@bryanchriswhite): Ensure validator key is always available via the in-memory network's keyring.
+	net := memnet.GetNetwork(t)
+	val := net.Validators[0]
+
+	args := []string{
+		fmt.Sprintf("--%s=true", flags.FlagOffline),
+		fmt.Sprintf("--%s=%d", flags.FlagAccountNumber, signerAccountNumber),
+		fmt.Sprintf("--%s=%d", flags.FlagSequence, memnet.NextAccountSequenceNumber(t)),
+
+		fmt.Sprintf("--%s=%s", flags.FlagFrom, val.Address.String()),
+		fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+		fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastSync),
+		fmt.Sprintf("--%s=%s", flags.FlagFees, types.NewCoins(types.NewCoin(net.Config.BondDenom, math.NewInt(10))).String()),
+	}
+	amount := types.NewCoins(types.NewCoin("stake", math.NewInt(200)))
+	responseRaw, err := testcli.MsgSendExec(clientCtx, val.Address, addr, amount, args...)
+	require.NoError(t, err)
+	var responseJSON map[string]interface{}
+	err = json.Unmarshal(responseRaw.Bytes(), &responseJSON)
+	require.NoError(t, err)
+	require.Equal(t, float64(0), responseJSON["code"], "code is not 0 in the response: %v", responseJSON)
+}
+
+// GetPreGeneratedAccountIterator returns the pre-generated account iterator associated
```

Comment:
Comment is mis-aligned with the function name (and functionality)

Reply:
:pray: Not sure what exactly happened here. :sweat_smile: :thinking:

Reply:
:pray: